python/hsfs/core/delta_engine.py

```diff
@@ -0,0 +1,176 @@
+#   You may obtain a copy of the License at
+#
+#       http://www.apache.org/licenses/LICENSE-2.0
+#
+#   Unless required by applicable law or agreed to in writing, software
+#   distributed under the License is distributed on an "AS IS" BASIS,
+#   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#   See the License for the specific language governing permissions and
+#   limitations under the License.
+#
+
+from hsfs import feature_group_commit, util
+from hsfs.core import feature_group_api
+
+try:
+    from delta.tables import DeltaTable
+except ImportError:
+    pass
+
+
+class DeltaEngine:
+    DELTA_SPARK_FORMAT = "delta"
+    DELTA_QUERY_TIME_TRAVEL_AS_OF_INSTANT = "timestampAsOf"
+
+    def __init__(
+        self,
+        feature_store_id,
+        feature_store_name,
+        feature_group,
+        spark_session,
+        spark_context,
+    ):
+        self._feature_group = feature_group
+        self._spark_context = spark_context
+        self._spark_session = spark_session
+        self._feature_store_id = feature_store_id
+        self._feature_store_name = feature_store_name
+
+        self._feature_group_api = feature_group_api.FeatureGroupApi()
+
+    def save_delta_fg(self, dataset, write_options, validation_id=None):
+        fg_commit = self._write_delta_dataset(dataset, write_options)
+        fg_commit.validation_id = validation_id
+        return self._feature_group_api.commit(self._feature_group, fg_commit)
+
+    def register_temporary_table(self, delta_fg_alias, read_options):
+        delta_options = self._setup_delta_read_opts(delta_fg_alias, read_options)
+        self._spark_session.read.format(self.DELTA_SPARK_FORMAT).options(
+            **delta_options
+        ).load(self._feature_group.location).createOrReplaceTempView(
+            delta_fg_alias.alias
+        )
+
+    def _setup_delta_read_opts(self, delta_fg_alias, read_options):
+        delta_options = {}
+        if delta_fg_alias.left_feature_group_end_timestamp is None and (
+            delta_fg_alias.left_feature_group_start_timestamp is None
+            or delta_fg_alias.left_feature_group_start_timestamp == 0
+        ):
+            # snapshot query latest state
+            delta_options = {}
+        elif (
+            delta_fg_alias.left_feature_group_end_timestamp is not None
+            and delta_fg_alias.left_feature_group_start_timestamp is None
+        ):
+            # snapshot query with end time
+            _delta_commit_end_time = util.get_delta_datestr_from_timestamp(
+                delta_fg_alias.left_feature_group_end_timestamp
+            )
+            delta_options = {
+                self.DELTA_QUERY_TIME_TRAVEL_AS_OF_INSTANT: _delta_commit_end_time,
+            }
+
+        if read_options:
+            delta_options.update(read_options)
+
+        return delta_options
+
+    def delete_record(self, delete_df, write_options):
+        pass
+
```

Comment:
Not used

Reply:
@bubriks - I think this is here to implement the delete records method support for delta - I think we can do it in a follow up PR.